src/core/text-buffer.cc

```diff
@@ -784,7 +787,7 @@ void TextBuffer::serialize_changes(Serializer &serializer) {
 
 bool TextBuffer::deserialize_changes(Deserializer &deserializer) {
   if (top_layer != base_layer || (base_layer->previous_layer != nullptr)) return false;
```

Comment:
Looks like C++2020 is going to be removing the != operator for shared pointers.
https://en.cppreference.com/w/cpp/memory/shared_ptr/operator_cmp
So might want to consider converting all the comparisons for compatibility with 2020